src/components/Select/Select/Select.tsx

```diff
@@ -57,7 +57,7 @@ export const Select = (props: TSingleSelectPropTypes): JSX.Element | null => {
   const [itemLabel, setItemLabel] = useState<string | null>('')
 
   const setCurrentSelectedLabel = useCallback(() => {
-    const label = options.find((item) => item.value === currentSelection)
+    const label = options?.find((item) => item.value === currentSelection)
```

Comment:
@LilitArz jan as i found `options` props is mandatory, and it means we do not need `?`

Reply:
when data is not still fetched options is null, and in that case we show loading, and these calculations should be done when options available.